app/controllers/leads_controller.rb

```diff
@@ -189,37 +187,61 @@ def destroy_couplet
     end
   end
 
-  # For deleting a couplet where one side has no children but the other might;
-  # the side with children is reparented.
-  def delete_couplet
+  # For deleting all children at a particular level of the key where at most one
+  # of the children has its own children; the grandchildren, if any, are
+  # reparented.
+  def delete_children
     respond_to do |format|
-      if @lead.destroy_couplet
+      if @lead.destroy_children
         format.json { head :no_content }
       else
-        @lead.errors.add(:delete, 'failed - is there a node redirecting to one of these?')
+        @lead.errors.add(:delete, 'failed!')
         format.json {
           render json: @lead.errors, status: :unprocessable_entity
         }
       end
     end
   end
 
-  # POST /leads/1/duplicate
+  # POST /leads/1/add_lead.json
+  def add_lead
+    parent = Lead.find(params[:id])
+    @lead = parent.children.create!
+  end
+
+  # POST /leads/1/duplicate.html
   def duplicate
```

Comment:
Can you clarify if this is intended to duplicate children?  If not then it feels like it's possibly doable with client side sending `def create` a filtered set of parameters?

Reply:
Yes, this duplicates all children of the lead it acts on.